src/api_main.py

```diff
@@ -0,0 +1,181 @@
+import pandas as pd
+from typing import Optional
+import os
+from fastapi import FastAPI, File, UploadFile, HTTPException
+from errors import ApkAnalyzerError
+from apk_analyzer import ApkAnalyzer
+from auxiliary_functions import Auxiliary
+from typing import List
+from errors import CacheException, OtherProvidersException, NonApkFileError, FileNotFoundError
+from redis_cache import RedisCacheManager
+from other_providers import OtherProviders
+import logging
+import asyncio
+from dotenv import load_dotenv
+
+if os.getenv("DOCKER_ENV"):
+    load_dotenv(".docker.env")
+else:
+    load_dotenv(".env")
+
+logging.basicConfig(filename='app.log',level=logging.DEBUG, format='%(asctime)s - %(levelname)s - %(message)s')
+
+app = FastAPI()
+
+COMMON_PERMISSIONS_PATH = os.getenv('COMMON_PERMISSIONS_PATH_FROM_SRC')
+
+VT_API_KEY = os.getenv('VT_API_KEY')
+VT_URL = os.getenv("VT_URL")
+KASPERSKY_API_KEY = os.getenv('KASPERSKY_API_KEY')
+
+most_common_permissions = pd.read_csv(COMMON_PERMISSIONS_PATH, header=None)
+permissions_list = most_common_permissions[0].tolist()[1:]
+
+# Retrieve environment variables
+HOST = os.getenv("REDIS_HOST")
+PORT = int(os.getenv("REDIS_PORT"))
+
+# initial class objects:
+apk_analyzer = ApkAnalyzer(permissions_list)
+redis_cache = RedisCacheManager(HOST, PORT)
```

Comment:
I want you to rely on an env variable, called: ENABLE_REDIS_CACHE, and use redis_cache_manager if it's on , and otherwise - use db_cache_manager